src/items/items.controller.ts

```diff
@@ -0,0 +1,54 @@
+//controller : client로부터의 요청 처리. GET 요청 -> 서비스호출 -> 데이터반환
+
+import { Controller, Get, Post, Param, Body, Delete } from '@nestjs/common';
+import { ItemsService } from './items.service';
+import { BuyItemDto } from './dto/buy-item.dto';
+import { EquipItemDto } from './dto/equip-item.dto';
+import { UnequipItemDto } from './dto/unequip-item.dto';
+
+@Controller('items') // '/items'경로 요청 처리
+export class ItemsController {
+  constructor(private readonly itemsService: ItemsService) {}
+
+  // GET /items 요청 처리
+  @Get()
+  async getAllItems() {
+    return await this.itemsService.getAllItems(); //service에서 아이템 목록을 받아서, 반환(return)
+  }
+
+  // POST /items/buy 요청 처리
+  @Post('buy')
+  async buyItem(@Body() buyItemDto: BuyItemDto) {
+    const { userId, itemId } = buyItemDto;
+    return await this.itemsService.buyItem(userId, itemId);
+  }
```

Comment:
```suggestion
  // POST /items/buy 요청 처리
  @Post('buy')
  async buyItem(@Body() buyItemDto: BuyItemDto) {
    return await this.itemsService.buyItem(buyItemDto);
  }
```
컨트롤러에서 dto를 구조분해할당 하는 것보단 service layer에서 처리하는게 나을 것 같습니다. 보통 컨트롤러에선 HTTP 요청과 응답을 처리하는 역할을 하고 service layer에서 비지니스 로직을 구현하는 편입니다.

equip쪽도 수정해주세용

Reply:
아 provider: []에 넣으면 인스턴스가 또 생기는군요.
다른곳의 메소드등을 사용하고 싶으면 import: []에 모듈을 넣어서 써야하는거구요. 알겠습니다.

Reply:
아, 중요한 점 잡아주셔서 감사합니다!!!!